lib/rules/media-feature-name-value-allowed-list/index.js

```diff
@@ -37,70 +40,98 @@ const rule = (primary) => {
 		}
 
 		root.walkAtRules(/^media$/i, (atRule) => {
-			mediaParser(atRule.params).walk(/^media-feature-expression$/i, (node) => {
-				if (!node.nodes) return;
-
-				const mediaFeatureRangeContext = isRangeContextMediaFeature(node.parent.value);
-
-				// Ignore boolean
-				if (!node.value.includes(':') && !mediaFeatureRangeContext) {
-					return;
-				}
-
-				const mediaFeatureNode = node.nodes.find((n) => n.type === 'media-feature');
-
-				if (mediaFeatureNode == null) throw new Error('A `media-feature` node must be present');
-
-				let mediaFeatureName;
-				let values;
+			parseMediaQuery(atRule).forEach((mediaQuery) => {
+				if (isMediaQueryInvalid(mediaQuery)) return;
+
+				mediaQuery.walk(
+					(entry) => {
+						if (!entry.state) return;
+
+						if (isMediaFeature(entry.node)) {
+							entry.state.mediaFeatureName = entry.node.getName();
+							entry.state.unprefixedMediaFeatureName = vendor.unprefixed(entry.node.getName());
+
+							return;
+						}
+
+						if (!isMediaFeatureValue(entry.node)) return;
+
+						if (!entry.state.mediaFeatureName || !entry.state.unprefixedMediaFeatureName) return;
+
+						const mediaFeatureName = entry.state.mediaFeatureName;
+						const unprefixedMediaFeatureName = entry.state.unprefixedMediaFeatureName;
```

Comment:
[suggest] It's possible to shorten the code a bit. How about this refactoring?

```suggestion
						const { mediaFeatureName, unprefixedMediaFeatureName } = entry.state;

						if (!mediaFeatureName || !unprefixedMediaFeatureName) return;
```